PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs

```diff
@@ -239,20 +247,25 @@ private static bool SearchThresholdExceeds(ISession session)
                                 p.Longitude,
                                 session)
                                 ).ToList();
+
             if (session.LogicSettings.UseGpxPathing)
             {
                 pokeStopes = pokeStopes.Where(p => LocationUtils.CalculateDistanceInMeters(p.Latitude, p.Longitude, session.Client.CurrentLatitude, session.Client.CurrentLongitude) < 40).ToList();
             }
+
             if (pokeStopes.Count == 1) return pokeStopes.FirstOrDefault();
 
-           if (session.LogicSettings.GymAllowed)
+            if (session.LogicSettings.GymAllowed)
             {
-                var gyms = pokeStopes.Where(x => x.Type == FortType.Gym &&
-                LocationUtils.CalculateDistanceInMeters(x.Latitude, x.Longitude, session.Client.CurrentLatitude, session.Client.CurrentLongitude) < session.LogicSettings.GymMaxDistance
-                && x.CooldownCompleteTimestampMs < DateTime.UtcNow.ToUnixTime());
+                var gyms = pokeStopes.Where(x => 
+                    x.Type == FortType.Gym &&
+                    LocationUtils.CalculateDistanceInMeters(x.Latitude, x.Longitude, session.Client.CurrentLatitude, session.Client.CurrentLongitude) < session.LogicSettings.GymMaxDistance
+                );
 
+                //TODO: Why Gym has higher priority?
                 if (gyms.Count() > 0) return gyms.FirstOrDefault();
```

Comment:
you have about 3% forts is pokestop. if you not set it as priority, you bot never hit any gym, because bot always walk away from gym. it also not visit all gym unless you set you farm area is 10m :)). you also remove logic add gyms from add session, that is the huge break, because gyms fort doesn't has follow 5 min cools down time, it you gym setting cooldown